### ISSUE TITLE
ES6 class You cannot rename this element

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript#8777]
_From @mikehaas763 on May 23, 2016 16:35_

- VSCode Version: 1.1.1
- OS Version: OS X El Capitan 10.11.5

Steps to Reproduce:

1. Create an ES6 class with a constructor

```javascript
export class AppEnvironment {
    constructor(environment) {
        this.environment = environment;
    }
}
```

2. right click the `environment` symbol that's part of the `this.environment` expression -> click rename symbol
3. Rename the symbol
4. "You cannot rename this element"

I'm aware that this probably doesn't work because it's just not implemented yet. Just want to bring it up to hope it gets implemented.

_Copied from original issue: Microsoft/vscode#6680_

**Comment by mhegazy** [comment]
This should work in a .js file, but not a .ts file.

**Comment by sandersn** [comment]
Symbols weren't being created for `this.x = ...` assignments in ES6 constructors. Fix is up at #8844
